### PR TITLE
improve: add API to fetch config and apply directly SDKCF-2483

### PR DESCRIPTION
**Pull request by donnie-jp**
# Description
Add API to fetch config and apply directly and a plist option for config to always be applied directly after fetch.

## Links
SDKCF-2483

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: RRemoteConfig/Bundle+Environment.swift [hunk 1]
```diff
@@ -44,4 +44,11 @@ extension Bundle: EnvironmentSetupProtocol {
         }
         return TimeInterval(truncating: number)
     }
+
+    func applyConfigDirectlyAfterFetch() -> Bool {
+        guard let applyDirectly = self.object(forInfoDictionaryKey: "RRCApplyConfigDirectlyAfterFetch") as? Bool else {
```
Comment (malayli) [inline RRemoteConfig/Bundle+Environment.swift:49]: `self.` is not needed

File: RRemoteConfig/ConfigCache.swift [hunk 2]
```diff
@@ -165,3 +179,10 @@ extension ConfigCache {
         }
     }
 }
+
+extension FileManager {
+    fileprivate class func getCacheDirectory() -> URL {
```
Comment (malayli) [inline RRemoteConfig/ConfigCache.swift:184]: you can replace this function by a computed property

File: RRemoteConfig/ConfigCache.swift [hunk 3]
```diff
@@ -87,23 +81,50 @@ internal class ConfigCache {
...
-                if verified {
-                    let dictionary = [
+    func fetchAndPollConfig() {
+        poller.start {
+            self.fetchConfig(applyDirectly: self.applyConfigDirectly) { _ in }
+        }
+    }
+
+    func fetchAndApplyConfig(completionHandler: @escaping (ConfigDictionary) -> Void) {
+        self.fetchConfig(applyDirectly: true) { (configModel) in
```
Comment (malayli) [inline RRemoteConfig/ConfigCache.swift:91]: same comment for `self.`, you can remove it

File: RRemoteConfig/ConfigCache.swift [hunk 4]
```diff
@@ -87,23 +81,50 @@ internal class ConfigCache {
...
+            completionHandler(configModel?.config ?? ConfigDictionary())
+        }
```
Comment (malayli) [inline RRemoteConfig/ConfigCache.swift:93]: You could change that by:
```
fetchConfig(applyDirectly: true) { completionHandler($0?.config ?? ConfigDictionary()) }
```

**Comment by imaginaris** [comment]
@donnie-jp What's up with this PR? Is it still valid?

**Comment by donnie-jp** [comment]
> @donnie-jp What's up with this PR? Is it still valid?

@imaginaris hmmm technically it is valid but as we aren't doing any work on Remote Config for the foreseeable future I'll just close this PR.

**Pull request closed without merging**